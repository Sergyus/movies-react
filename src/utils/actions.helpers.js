export const createAsyncFnRequest = requestName => ({
  type: requestName,
});

export const createAsyncFnFailure = (requestName, error) => ({
  type: requestName,
  error: error.message,
});
