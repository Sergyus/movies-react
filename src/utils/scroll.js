const getScrollDownPercentage = ({ scrollHeight, scrollPos, clientHeight }) => {
  const currentPosition = scrollPos + clientHeight;
  const percentageScrolled = currentPosition / scrollHeight;
  return percentageScrolled;
};

export default getScrollDownPercentage;
