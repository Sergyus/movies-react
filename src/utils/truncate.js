const truncate = (description = '', maxLength) => (
  description.length && description.length > maxLength ? `${description.slice(0, maxLength)}...` : description
);

export default truncate;
