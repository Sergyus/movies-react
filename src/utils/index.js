import {
  getTopMovies, searchMovies, getGenres, getMoviesByGenre,
  getLatestMovies, getUpcomingMovies, getMovieVideo,
} from './api';
import { getMovies } from './tbdbApiService';
import truncate from './truncate';
import { useFetchOnScroll } from './useFetchOnScroll';
import useBodyScrollLock from './useBodyScrollLock';
import useOnClickOutside from './useOnClickOutside';
import { createAsyncFnFailure, createAsyncFnRequest } from './actions.helpers';

export {
  getTopMovies, searchMovies, getGenres, getMoviesByGenre,
  getLatestMovies, getUpcomingMovies, getMovieVideo,
  truncate, useBodyScrollLock, useOnClickOutside, getMovies,
  createAsyncFnFailure, createAsyncFnRequest, useFetchOnScroll,
};
