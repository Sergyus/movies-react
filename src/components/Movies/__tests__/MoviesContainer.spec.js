import React from 'react';
import { create, act } from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import ReactDOM from 'react-dom';
import MoviesContainer from '..';
import { results } from '../../../assets/json/popular.json';
import { genres } from '../../../assets/json/genres.json';
import { getMovies } from '../../../utils/tbdbApiService';

global.fetch = require('jest-fetch-mock');

describe('MoviesContainer component renders correctly', () => {
  beforeAll(() => {
    ReactDOM.createPortal = jest.fn(element => element);
  });
  afterEach(() => {
    ReactDOM.createPortal.mockClear();
  });
  const middlewares = [thunk];
  const mockStore = configureStore(middlewares);
  const movies = getMovies(results, genres);
  it('MoviesContainer grid render and unmount correctly', () => {
    const initialState = {
      app: {
        movies,
        genres,
        isLoading: true,
        currentPage: 1,
        error: false,
        filter: 'Trending',
        currentGenre: '',
        currentSearchQuery: '',
        videoKey: '',
        currentTotalPages: 1,
      },
      ui: {
        layout: 'grid',
      },
    };
    const store = mockStore(initialState);
    let tree;
    act(() => {
      tree = create(
        <Provider store={store}>
          <MoviesContainer />
        </Provider>,
      );
    });
    expect(tree.toJSON()).toMatchSnapshot();
    tree.unmount();
  });
  it('MoviesContainer table render correctly', () => {
    const initialState = {
      app: {
        movies,
        genres,
        isLoading: true,
        currentPage: 1,
        error: false,
        filter: 'Trending',
        currentGenre: '',
        currentSearchQuery: '',
        videoKey: '',
        currentTotalPages: 1,
      },
      ui: {
        layout: 'table',
      },
    };
    const store = mockStore(initialState);
    const tree = create(
      <Provider store={store}>
        <MoviesContainer />
      </Provider>,
    );
    const button = tree.root.findAllByProps({ 'aria-label': 'Watch Now' })[0];
    const mockFn = jest.fn(button.props.onClick);
    act(() => {
      mockFn();
    });
    expect(mockFn).toHaveBeenCalled();
    expect(tree.toJSON()).toMatchSnapshot();
  });
  it('MoviesContainer without layout', () => {
    const initialState = {
      app: {
        movies,
        genres,
        isLoading: true,
        currentPage: 1,
        error: false,
        filter: 'Trending',
        currentGenre: '',
        currentSearchQuery: '',
        videoKey: '',
        currentTotalPages: 1,
      },
      ui: {
        layout: '',
      },
    };
    const store = mockStore(initialState);
    const tree = create(
      <Provider store={store}>
        <MoviesContainer />
      </Provider>,
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
  it('MoviesContainer without movies table render correctly', () => {
    const initialState = {
      app: {
        movies: [],
        genres,
        isLoading: true,
        currentPage: 1,
        error: false,
        filter: 'Trending',
        currentGenre: '',
        currentSearchQuery: '',
        videoKey: '',
        currentTotalPages: 1,
      },
      ui: {
        layout: 'table',
      },
    };
    const store = mockStore(initialState);
    let tree;
    act(() => {
      tree = create(
        <Provider store={store}>
          <MoviesContainer />
        </Provider>,
      );
    });
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
