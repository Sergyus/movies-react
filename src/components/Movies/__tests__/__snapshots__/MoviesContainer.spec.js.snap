// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesContainer component renders correctly MoviesContainer grid render and unmount correctly 1`] = `
Array [
  <main
    className="container gridMovieList"
  >
    <article
      className="card"
    >
      <img
        alt="Poster"
        height="70%"
        src={
          Object {
            "process": [Function],
          }
        }
      />
      <div
        className="onHover"
      >
        <button
          aria-label="Watch"
          className="btnWatch"
          onClick={[Function]}
          type="button"
        >
          <span>
             Watch Now 
          </span>
        </button>
        <button
          aria-label="View Info"
          className="btn ghost"
          onClick={[Function]}
          type="button"
        >
          View Info
        </button>
      </div>
      <div
        className="info"
      >
        <div>
          <h3
            className="title"
          >
            T-34
          </h3>
          <p />
        </div>
      </div>
    </article>
    <article
      className="card"
    >
      <img
        alt="Poster"
        height="70%"
        src={
          Object {
            "process": [Function],
          }
        }
      />
      <div
        className="onHover"
      >
        <button
          aria-label="Watch"
          className="btnWatch"
          onClick={[Function]}
          type="button"
        >
          <span>
             Watch Now 
          </span>
        </button>
        <button
          aria-label="View Info"
          className="btn ghost"
          onClick={[Function]}
          type="button"
        >
          View Info
        </button>
      </div>
      <div
        className="info"
      >
        <div>
          <h3
            className="title"
          >
            Glass
          </h3>
          <p />
        </div>
      </div>
    </article>
    <article
      className="card"
    >
      <img
        alt="Poster"
        height="70%"
        src={
          Object {
            "process": [Function],
          }
        }
      />
      <div
        className="onHover"
      >
        <button
          aria-label="Watch"
          className="btnWatch"
          onClick={[Function]}
          type="button"
        >
          <span>
             Watch Now 
          </span>
        </button>
        <button
          aria-label="View Info"
          className="btn ghost"
          onClick={[Function]}
          type="button"
        >
          View Info
        </button>
      </div>
      <div
        className="info"
      >
        <div>
          <h3
            className="title"
          >
            Robin Hood
          </h3>
          <p />
        </div>
      </div>
    </article>
    <article
      className="card"
    >
      <img
        alt="Poster"
        height="70%"
        src={
          Object {
            "process": [Function],
          }
        }
      />
      <div
        className="onHover"
      >
        <button
          aria-label="Watch"
          className="btnWatch"
          onClick={[Function]}
          type="button"
        >
          <span>
             Watch Now 
          </span>
        </button>
        <button
          aria-label="View Info"
          className="btn ghost"
          onClick={[Function]}
          type="button"
        >
          View Info
        </button>
      </div>
      <div
        className="info"
      >
        <div>
          <h3
            className="title"
          >
            Bohemian Rhapsody
          </h3>
          <p />
        </div>
      </div>
    </article>
    <article
      className="card"
    >
      <img
        alt="Poster"
        height="70%"
        src={
          Object {
            "process": [Function],
          }
        }
      />
      <div
        className="onHover"
      >
        <button
          aria-label="Watch"
          className="btnWatch"
          onClick={[Function]}
          type="button"
        >
          <span>
             Watch Now 
          </span>
        </button>
        <button
          aria-label="View Info"
          className="btn ghost"
          onClick={[Function]}
          type="button"
        >
          View Info
        </button>
      </div>
      <div
        className="info"
      >
        <div>
          <h3
            className="title"
          >
            Overlord
          </h3>
          <p />
        </div>
      </div>
    </article>
    <article
      className="card"
    >
      <img
        alt="Poster"
        height="70%"
        src={
          Object {
            "process": [Function],
          }
        }
      />
      <div
        className="onHover"
      >
        <button
          aria-label="Watch"
          className="btnWatch"
          onClick={[Function]}
          type="button"
        >
          <span>
             Watch Now 
          </span>
        </button>
        <button
          aria-label="View Info"
          className="btn ghost"
          onClick={[Function]}
          type="button"
        >
          View Info
        </button>
      </div>
      <div
        className="info"
      >
        <div>
          <h3
            className="title"
          >
            Escape Room
          </h3>
          <p />
        </div>
      </div>
    </article>
    <article
      className="card"
    >
      <img
        alt="Poster"
        height="70%"
        src={
          Object {
            "process": [Function],
          }
        }
      />
      <div
        className="onHover"
      >
        <button
          aria-label="Watch"
          className="btnWatch"
          onClick={[Function]}
          type="button"
        >
          <span>
             Watch Now 
          </span>
        </button>
        <button
          aria-label="View Info"
          className="btn ghost"
          onClick={[Function]}
          type="button"
        >
          View Info
        </button>
      </div>
      <div
        className="info"
      >
        <div>
          <h3
            className="title"
          >
            Aquaman
          </h3>
          <p />
        </div>
      </div>
    </article>
    <article
      className="card"
    >
      <img
        alt="Poster"
        height="70%"
        src={
          Object {
            "process": [Function],
          }
        }
      />
      <div
        className="onHover"
      >
        <button
          aria-label="Watch"
          className="btnWatch"
          onClick={[Function]}
          type="button"
        >
          <span>
             Watch Now 
          </span>
        </button>
        <button
          aria-label="View Info"
          className="btn ghost"
          onClick={[Function]}
          type="button"
        >
          View Info
        </button>
      </div>
      <div
        className="info"
      >
        <div>
          <h3
            className="title"
          >
            Mortal Engines
          </h3>
          <p />
        </div>
      </div>
    </article>
    <article
      className="card"
    >
      <img
        alt="Poster"
        height="70%"
        src={
          Object {
            "process": [Function],
          }
        }
      />
      <div
        className="onHover"
      >
        <button
          aria-label="Watch"
          className="btnWatch"
          onClick={[Function]}
          type="button"
        >
          <span>
             Watch Now 
          </span>
        </button>
        <button
          aria-label="View Info"
          className="btn ghost"
          onClick={[Function]}
          type="button"
        >
          View Info
        </button>
      </div>
      <div
        className="info"
      >
        <div>
          <h3
            className="title"
          >
            Creed II
          </h3>
          <p />
        </div>
      </div>
    </article>
    <article
      className="card"
    >
      <img
        alt="Poster"
        height="70%"
        src={
          Object {
            "process": [Function],
          }
        }
      />
      <div
        className="onHover"
      >
        <button
          aria-label="Watch"
          className="btnWatch"
          onClick={[Function]}
          type="button"
        >
          <span>
             Watch Now 
          </span>
        </button>
        <button
          aria-label="View Info"
          className="btn ghost"
          onClick={[Function]}
          type="button"
        >
          View Info
        </button>
      </div>
      <div
        className="info"
      >
        <div>
          <h3
            className="title"
          >
            How to Train Your Dragon: The Hidden World
          </h3>
          <p />
        </div>
      </div>
    </article>
    <article
      className="card"
    >
      <img
        alt="Poster"
        height="70%"
        src={
          Object {
            "process": [Function],
          }
        }
      />
      <div
        className="onHover"
      >
        <button
          aria-label="Watch"
          className="btnWatch"
          onClick={[Function]}
          type="button"
        >
          <span>
             Watch Now 
          </span>
        </button>
        <button
          aria-label="View Info"
          className="btn ghost"
          onClick={[Function]}
          type="button"
        >
          View Info
        </button>
      </div>
      <div
        className="info"
      >
        <div>
          <h3
            className="title"
          >
            A Star Is Born
          </h3>
          <p />
        </div>
      </div>
    </article>
    <article
      className="card"
    >
      <img
        alt="Poster"
        height="70%"
        src={
          Object {
            "process": [Function],
          }
        }
      />
      <div
        className="onHover"
      >
        <button
          aria-label="Watch"
          className="btnWatch"
          onClick={[Function]}
          type="button"
        >
          <span>
             Watch Now 
          </span>
        </button>
        <button
          aria-label="View Info"
          className="btn ghost"
          onClick={[Function]}
          type="button"
        >
          View Info
        </button>
      </div>
      <div
        className="info"
      >
        <div>
          <h3
            className="title"
          >
            The Mule
          </h3>
          <p />
        </div>
      </div>
    </article>
    <article
      className="card"
    >
      <img
        alt="Poster"
        height="70%"
        src={
          Object {
            "process": [Function],
          }
        }
      />
      <div
        className="onHover"
      >
        <button
          aria-label="Watch"
          className="btnWatch"
          onClick={[Function]}
          type="button"
        >
          <span>
             Watch Now 
          </span>
        </button>
        <button
          aria-label="View Info"
          className="btn ghost"
          onClick={[Function]}
          type="button"
        >
          View Info
        </button>
      </div>
      <div
        className="info"
      >
        <div>
          <h3
            className="title"
          >
            Avengers: Infinity War
          </h3>
          <p />
        </div>
      </div>
    </article>
    <article
      className="card"
    >
      <img
        alt="Poster"
        height="70%"
        src={
          Object {
            "process": [Function],
          }
        }
      />
      <div
        className="onHover"
      >
        <button
          aria-label="Watch"
          className="btnWatch"
          onClick={[Function]}
          type="button"
        >
          <span>
             Watch Now 
          </span>
        </button>
        <button
          aria-label="View Info"
          className="btn ghost"
          onClick={[Function]}
          type="button"
        >
          View Info
        </button>
      </div>
      <div
        className="info"
      >
        <div>
          <h3
            className="title"
          >
            Bumblebee
          </h3>
          <p />
        </div>
      </div>
    </article>
    <article
      className="card"
    >
      <img
        alt="Poster"
        height="70%"
        src={
          Object {
            "process": [Function],
          }
        }
      />
      <div
        className="onHover"
      >
        <button
          aria-label="Watch"
          className="btnWatch"
          onClick={[Function]}
          type="button"
        >
          <span>
             Watch Now 
          </span>
        </button>
        <button
          aria-label="View Info"
          className="btn ghost"
          onClick={[Function]}
          type="button"
        >
          View Info
        </button>
      </div>
      <div
        className="info"
      >
        <div>
          <h3
            className="title"
          >
            Widows
          </h3>
          <p />
        </div>
      </div>
    </article>
    <article
      className="card"
    >
      <img
        alt="Poster"
        height="70%"
        src={
          Object {
            "process": [Function],
          }
        }
      />
      <div
        className="onHover"
      >
        <button
          aria-label="Watch"
          className="btnWatch"
          onClick={[Function]}
          type="button"
        >
          <span>
             Watch Now 
          </span>
        </button>
        <button
          aria-label="View Info"
          className="btn ghost"
          onClick={[Function]}
          type="button"
        >
          View Info
        </button>
      </div>
      <div
        className="info"
      >
        <div>
          <h3
            className="title"
          >
            Serenity
          </h3>
          <p />
        </div>
      </div>
    </article>
    <article
      className="card"
    >
      <img
        alt="Poster"
        height="70%"
        src={
          Object {
            "process": [Function],
          }
        }
      />
      <div
        className="onHover"
      >
        <button
          aria-label="Watch"
          className="btnWatch"
          onClick={[Function]}
          type="button"
        >
          <span>
             Watch Now 
          </span>
        </button>
        <button
          aria-label="View Info"
          className="btn ghost"
          onClick={[Function]}
          type="button"
        >
          View Info
        </button>
      </div>
      <div
        className="info"
      >
        <div>
          <h3
            className="title"
          >
            Venom
          </h3>
          <p />
        </div>
      </div>
    </article>
    <article
      className="card"
    >
      <img
        alt="Poster"
        height="70%"
        src={
          Object {
            "process": [Function],
          }
        }
      />
      <div
        className="onHover"
      >
        <button
          aria-label="Watch"
          className="btnWatch"
          onClick={[Function]}
          type="button"
        >
          <span>
             Watch Now 
          </span>
        </button>
        <button
          aria-label="View Info"
          className="btn ghost"
          onClick={[Function]}
          type="button"
        >
          View Info
        </button>
      </div>
      <div
        className="info"
      >
        <div>
          <h3
            className="title"
          >
            Can You Ever Forgive Me?
          </h3>
          <p />
        </div>
      </div>
    </article>
    <article
      className="card"
    >
      <img
        alt="Poster"
        height="70%"
        src={
          Object {
            "process": [Function],
          }
        }
      />
      <div
        className="onHover"
      >
        <button
          aria-label="Watch"
          className="btnWatch"
          onClick={[Function]}
          type="button"
        >
          <span>
             Watch Now 
          </span>
        </button>
        <button
          aria-label="View Info"
          className="btn ghost"
          onClick={[Function]}
          type="button"
        >
          View Info
        </button>
      </div>
      <div
        className="info"
      >
        <div>
          <h3
            className="title"
          >
            Fate/stay night: Heaven’s Feel II. lost butterfly
          </h3>
          <p />
        </div>
      </div>
    </article>
    <article
      className="card"
    >
      <img
        alt="Poster"
        height="70%"
        src={
          Object {
            "process": [Function],
          }
        }
      />
      <div
        className="onHover"
      >
        <button
          aria-label="Watch"
          className="btnWatch"
          onClick={[Function]}
          type="button"
        >
          <span>
             Watch Now 
          </span>
        </button>
        <button
          aria-label="View Info"
          className="btn ghost"
          onClick={[Function]}
          type="button"
        >
          View Info
        </button>
      </div>
      <div
        className="info"
      >
        <div>
          <h3
            className="title"
          >
            LelleBelle
          </h3>
          <p />
        </div>
      </div>
    </article>
  </main>,
  <div
    className="spinner"
  >
    <div
      className="ellipsis"
    >
      <div />
      <div />
      <div />
      <div />
    </div>
    <p
      className="title"
    >
      Loading
    </p>
  </div>,
]
`;

exports[`MoviesContainer component renders correctly MoviesContainer table render correctly 1`] = `
Array [
  <main
    className="container tableMovieList"
  >
    <article
      className="card"
    >
      <img
        alt="Poster"
        src={
          Object {
            "process": [Function],
          }
        }
      />
      <div
        className="info"
      >
        <h2
          className="title"
        >
          T-34
        </h2>
        <p
          className="overview"
        >
          1941 - WWii. the second lieutenant Nikolai ivushkin, commander of a t-34, engages in an unequal battle against the tank ace Klaus Jager in a battle near moscow. His mission is more of a suicide - to destroy a dozen german tanks, all by himself. that said, luck does favour the bold. He wins the battle, barely survives, but loses his tank and lands himself in captivity for three long years... there was little to no chance for ivushkin and Jager to meet again, but the war knows how to throw a curve...
        </p>
        <div
          className="rating"
        />
        <button
          aria-label="Watch Now"
          className="btn primary"
          onClick={[Function]}
          type="button"
        >
          Watch Now
        </button>
      </div>
    </article>
    <article
      className="card"
    >
      <img
        alt="Poster"
        src={
          Object {
            "process": [Function],
          }
        }
      />
      <div
        className="info"
      >
        <h2
          className="title"
        >
          Glass
        </h2>
        <p
          className="overview"
        >
          In a series of escalating encounters, security guard David Dunn uses his supernatural abilities to track Kevin Wendell Crumb, a disturbed man who has twenty-four personalities. Meanwhile, the shadowy presence of Elijah Price emerges as an orchestrator who holds secrets critical to both men.
        </p>
        <div
          className="rating"
        />
        <button
          aria-label="Watch Now"
          className="btn primary"
          onClick={[Function]}
          type="button"
        >
          Watch Now
        </button>
      </div>
    </article>
    <article
      className="card"
    >
      <img
        alt="Poster"
        src={
          Object {
            "process": [Function],
          }
        }
      />
      <div
        className="info"
      >
        <h2
          className="title"
        >
          Robin Hood
        </h2>
        <p
          className="overview"
        >
          A war-hardened Crusader and his Moorish commander mount an audacious revolt against the corrupt English crown.
        </p>
        <div
          className="rating"
        />
        <button
          aria-label="Watch Now"
          className="btn primary"
          onClick={[Function]}
          type="button"
        >
          Watch Now
        </button>
      </div>
    </article>
    <article
      className="card"
    >
      <img
        alt="Poster"
        src={
          Object {
            "process": [Function],
          }
        }
      />
      <div
        className="info"
      >
        <h2
          className="title"
        >
          Bohemian Rhapsody
        </h2>
        <p
          className="overview"
        >
          Singer Freddie Mercury, guitarist Brian May, drummer Roger Taylor and bass guitarist John Deacon take the music world by storm when they form the rock 'n' roll band Queen in 1970. Hit songs become instant classics. When Mercury's increasingly wild lifestyle starts to spiral out of control, Queen soon faces its greatest challenge yet – finding a way to keep the band together amid the success and excess.
        </p>
        <div
          className="rating"
        />
        <button
          aria-label="Watch Now"
          className="btn primary"
          onClick={[Function]}
          type="button"
        >
          Watch Now
        </button>
      </div>
    </article>
    <article
      className="card"
    >
      <img
        alt="Poster"
        src={
          Object {
            "process": [Function],
          }
        }
      />
      <div
        className="info"
      >
        <h2
          className="title"
        >
          Overlord
        </h2>
        <p
          className="overview"
        >
          France, World War II, June 1944. On the eve of D-Day, some American paratroopers fall behind enemy lines after their plane crashes during a mission consisting of destroying a German radio tower in Cielblanc, a small village near the beaches of Normandy. After reaching their target, the surviving paratroopers realize that, in addition to fighting Nazi troops that patrol the village, they also must fight against something else.
        </p>
        <div
          className="rating"
        />
        <button
          aria-label="Watch Now"
          className="btn primary"
          onClick={[Function]}
          type="button"
        >
          Watch Now
        </button>
      </div>
    </article>
    <article
      className="card"
    >
      <img
        alt="Poster"
        src={
          Object {
            "process": [Function],
          }
        }
      />
      <div
        className="info"
      >
        <h2
          className="title"
        >
          Escape Room
        </h2>
        <p
          className="overview"
        >
          Six strangers find themselves in circumstances beyond their control, and must use their wits to survive.
        </p>
        <div
          className="rating"
        />
        <button
          aria-label="Watch Now"
          className="btn primary"
          onClick={[Function]}
          type="button"
        >
          Watch Now
        </button>
      </div>
    </article>
    <article
      className="card"
    >
      <img
        alt="Poster"
        src={
          Object {
            "process": [Function],
          }
        }
      />
      <div
        className="info"
      >
        <h2
          className="title"
        >
          Aquaman
        </h2>
        <p
          className="overview"
        >
          Once home to the most advanced civilization on Earth, the city of Atlantis is now an underwater kingdom ruled by the power-hungry King Orm. With a vast army at his disposal, Orm plans to conquer the remaining oceanic people -- and then the surface world. Standing in his way is Aquaman, Orm's half-human, half-Atlantean brother and true heir to the throne. With help from royal counselor Vulko, Aquaman must retrieve the legendary Trident of Atlan and embrace his destiny as protector of the deep.
        </p>
        <div
          className="rating"
        />
        <button
          aria-label="Watch Now"
          className="btn primary"
          onClick={[Function]}
          type="button"
        >
          Watch Now
        </button>
      </div>
    </article>
    <article
      className="card"
    >
      <img
        alt="Poster"
        src={
          Object {
            "process": [Function],
          }
        }
      />
      <div
        className="info"
      >
        <h2
          className="title"
        >
          Mortal Engines
        </h2>
        <p
          className="overview"
        >
          Set in a world many thousands of years in the future. Earth’s cities now roam the globe on huge wheels, devouring each other in a struggle for ever diminishing resources. On one of these massive Traction Cities, Tom Natsworthy has an unexpected encounter with a mysterious young woman from the Outlands who will change the course of his life forever.
        </p>
        <div
          className="rating"
        />
        <button
          aria-label="Watch Now"
          className="btn primary"
          onClick={[Function]}
          type="button"
        >
          Watch Now
        </button>
      </div>
    </article>
    <article
      className="card"
    >
      <img
        alt="Poster"
        src={
          Object {
            "process": [Function],
          }
        }
      />
      <div
        className="info"
      >
        <h2
          className="title"
        >
          Creed II
        </h2>
        <p
          className="overview"
        >
          Follows Adonis Creed's life inside and outside of the ring as he deals with new found fame, issues with his family, and his continuing quest to become a champion.
        </p>
        <div
          className="rating"
        />
        <button
          aria-label="Watch Now"
          className="btn primary"
          onClick={[Function]}
          type="button"
        >
          Watch Now
        </button>
      </div>
    </article>
    <article
      className="card"
    >
      <img
        alt="Poster"
        src={
          Object {
            "process": [Function],
          }
        }
      />
      <div
        className="info"
      >
        <h2
          className="title"
        >
          How to Train Your Dragon: The Hidden World
        </h2>
        <p
          className="overview"
        >
          As Hiccup fulfills his dream of creating a peaceful dragon utopia, Toothless’ discovery of an untamed, elusive mate draws the Night Fury away.  When danger mounts at home and Hiccup’s reign as village chief is tested, both dragon and rider must make impossible decisions to save their kind.
        </p>
        <div
          className="rating"
        />
        <button
          aria-label="Watch Now"
          className="btn primary"
          onClick={[Function]}
          type="button"
        >
          Watch Now
        </button>
      </div>
    </article>
    <article
      className="card"
    >
      <img
        alt="Poster"
        src={
          Object {
            "process": [Function],
          }
        }
      />
      <div
        className="info"
      >
        <h2
          className="title"
        >
          A Star Is Born
        </h2>
        <p
          className="overview"
        >
          Seasoned musician Jackson Maine discovers — and falls in love with — struggling artist Ally. She has just about given up on her dream to make it big as a singer — until Jack coaxes her into the spotlight. But even as Ally's career takes off, the personal side of their relationship is breaking down, as Jack fights an ongoing battle with his own internal demons.
        </p>
        <div
          className="rating"
        />
        <button
          aria-label="Watch Now"
          className="btn primary"
          onClick={[Function]}
          type="button"
        >
          Watch Now
        </button>
      </div>
    </article>
    <article
      className="card"
    >
      <img
        alt="Poster"
        src={
          Object {
            "process": [Function],
          }
        }
      />
      <div
        className="info"
      >
        <h2
          className="title"
        >
          The Mule
        </h2>
        <p
          className="overview"
        >
          A 90 year old horticulturalist and Korean War veteran is caught transporting $3 million worth of cocaine through Illinois for a Mexican drug cartel.
        </p>
        <div
          className="rating"
        />
        <button
          aria-label="Watch Now"
          className="btn primary"
          onClick={[Function]}
          type="button"
        >
          Watch Now
        </button>
      </div>
    </article>
    <article
      className="card"
    >
      <img
        alt="Poster"
        src={
          Object {
            "process": [Function],
          }
        }
      />
      <div
        className="info"
      >
        <h2
          className="title"
        >
          Avengers: Infinity War
        </h2>
        <p
          className="overview"
        >
          As the Avengers and their allies have continued to protect the world from threats too large for any one hero to handle, a new danger has emerged from the cosmic shadows: Thanos. A despot of intergalactic infamy, his goal is to collect all six Infinity Stones, artifacts of unimaginable power, and use them to inflict his twisted will on all of reality. Everything the Avengers have fought for has led up to this moment - the fate of Earth and existence itself has never been more uncertain.
        </p>
        <div
          className="rating"
        />
        <button
          aria-label="Watch Now"
          className="btn primary"
          onClick={[Function]}
          type="button"
        >
          Watch Now
        </button>
      </div>
    </article>
    <article
      className="card"
    >
      <img
        alt="Poster"
        src={
          Object {
            "process": [Function],
          }
        }
      />
      <div
        className="info"
      >
        <h2
          className="title"
        >
          Bumblebee
        </h2>
        <p
          className="overview"
        >
          On the run in the year 1987, Bumblebee finds refuge in a junkyard in a small Californian beach town. Charlie, on the cusp of turning 18 and trying to find her place in the world, discovers Bumblebee, battle-scarred and broken.  When Charlie revives him, she quickly learns this is no ordinary yellow VW bug.
        </p>
        <div
          className="rating"
        />
        <button
          aria-label="Watch Now"
          className="btn primary"
          onClick={[Function]}
          type="button"
        >
          Watch Now
        </button>
      </div>
    </article>
    <article
      className="card"
    >
      <img
        alt="Poster"
        src={
          Object {
            "process": [Function],
          }
        }
      />
      <div
        className="info"
      >
        <h2
          className="title"
        >
          Widows
        </h2>
        <p
          className="overview"
        >
          A police shootout leaves four thieves dead during an explosive armed robbery attempt in Chicago. Their widows have nothing in common except a debt left behind by their spouses' criminal activities. Hoping to forge a future on their own terms, they join forces to pull off a heist.
        </p>
        <div
          className="rating"
        />
        <button
          aria-label="Watch Now"
          className="btn primary"
          onClick={[Function]}
          type="button"
        >
          Watch Now
        </button>
      </div>
    </article>
    <article
      className="card"
    >
      <img
        alt="Poster"
        src={
          Object {
            "process": [Function],
          }
        }
      />
      <div
        className="info"
      >
        <h2
          className="title"
        >
          Serenity
        </h2>
        <p
          className="overview"
        >
          Baker Dill is a fishing boat captain leading tours off a tranquil, tropical enclave called Plymouth Island. His quiet life is shattered, however, when his ex-wife Karen tracks him down with a desperate plea for help.
        </p>
        <div
          className="rating"
        />
        <button
          aria-label="Watch Now"
          className="btn primary"
          onClick={[Function]}
          type="button"
        >
          Watch Now
        </button>
      </div>
    </article>
    <article
      className="card"
    >
      <img
        alt="Poster"
        src={
          Object {
            "process": [Function],
          }
        }
      />
      <div
        className="info"
      >
        <h2
          className="title"
        >
          Venom
        </h2>
        <p
          className="overview"
        >
          Investigative journalist Eddie Brock attempts a comeback following a scandal, but accidentally becomes the host of Venom, a violent, super powerful alien symbiote. Soon, he must rely on his newfound powers to protect the world from a shadowy organization looking for a symbiote of their own.
        </p>
        <div
          className="rating"
        />
        <button
          aria-label="Watch Now"
          className="btn primary"
          onClick={[Function]}
          type="button"
        >
          Watch Now
        </button>
      </div>
    </article>
    <article
      className="card"
    >
      <img
        alt="Poster"
        src={
          Object {
            "process": [Function],
          }
        }
      />
      <div
        className="info"
      >
        <h2
          className="title"
        >
          Can You Ever Forgive Me?
        </h2>
        <p
          className="overview"
        >
          When bestselling celebrity biographer Lee Israel is no longer able to get published because she has fallen out of step with current tastes, she turns her art form to deception, abetted by her loyal friend Jack.  It is a story of two difficult, socially inept, people somehow becoming partners-in-crime in forgeries: one creating and the other selling it.
        </p>
        <div
          className="rating"
        />
        <button
          aria-label="Watch Now"
          className="btn primary"
          onClick={[Function]}
          type="button"
        >
          Watch Now
        </button>
      </div>
    </article>
    <article
      className="card"
    >
      <img
        alt="Poster"
        src={
          Object {
            "process": [Function],
          }
        }
      />
      <div
        className="info"
      >
        <h2
          className="title"
        >
          Fate/stay night: Heaven’s Feel II. lost butterfly
        </h2>
        <p
          className="overview"
        >
          Theatrical-release adaptation of the visual novel "Fate/stay night", following the third and final route. (Part 2 of a trilogy.)
        </p>
        <div
          className="rating"
        />
        <button
          aria-label="Watch Now"
          className="btn primary"
          onClick={[Function]}
          type="button"
        >
          Watch Now
        </button>
      </div>
    </article>
    <article
      className="card"
    >
      <img
        alt="Poster"
        src={
          Object {
            "process": [Function],
          }
        }
      />
      <div
        className="info"
      >
        <h2
          className="title"
        >
          LelleBelle
        </h2>
        <p
          className="overview"
        >
          19-year old Belle practices playing the violin diligently, but is unable to impart her music with a sensitive undertone. Her life changes when she discovers that sexual desire stimulates passion in her music.
        </p>
        <div
          className="rating"
        />
        <button
          aria-label="Watch Now"
          className="btn primary"
          onClick={[Function]}
          type="button"
        >
          Watch Now
        </button>
      </div>
    </article>
  </main>,
  <div
    className="videoContainer"
  >
    <div
      className="spinner"
    >
      <div
        className="ellipsis"
      >
        <div />
        <div />
        <div />
        <div />
      </div>
      <p
        className="title"
      >
        Loading
      </p>
    </div>
  </div>,
  <div
    className="spinner"
  >
    <div
      className="ellipsis"
    >
      <div />
      <div />
      <div />
      <div />
    </div>
    <p
      className="title"
    >
      Loading
    </p>
  </div>,
]
`;

exports[`MoviesContainer component renders correctly MoviesContainer without layout 1`] = `
Array [
  <main
    className="container"
  />,
  <div
    className="spinner"
  >
    <div
      className="ellipsis"
    >
      <div />
      <div />
      <div />
      <div />
    </div>
    <p
      className="title"
    >
      Loading
    </p>
  </div>,
]
`;

exports[`MoviesContainer component renders correctly MoviesContainer without movies table render correctly 1`] = `
Array [
  <main
    className="container"
  >
    <h2>
      Sorry, no movies found.
    </h2>
  </main>,
  <div
    className="spinner"
  >
    <div
      className="ellipsis"
    >
      <div />
      <div />
      <div />
      <div />
    </div>
    <p
      className="title"
    >
      Loading
    </p>
  </div>,
]
`;
